toonz/sources/toonz/ocaio.cpp

```diff
@@ -534,4 +535,448 @@ void ExportOCACommand::execute() {
+      // else the group is only used as a way to group the layers in the UI of the application
+      importOcaLayer(layer.toObject());
+    }
+
+    TApp::instance()->getCurrentLevel()->setLevel(
+        level->getSimpleLevel());  // selects the last created level
+    TApp::instance()->getCurrentLevel()->notifyLevelChange();
+    TApp::instance()->getCurrentScene()->notifyCastChange();
+
+  } else {
+    DVGui::warning(QObject::tr("importOcaLayer skipping unimplemented layer type : ") +
+        jsonLayer["type"].toString() + " : " + jsonLayer["name"].toString()
+    );
+  }
+}
+
+/// <summary>
+/// creates frame from OCA frame
+/// see OCA frame specs
+/// https://github.com/RxLaboratory/OCA/blob/master/src-docs/docs/specs/frame.md
+/// UNUSED properties :
+/// - meta: metadata...
+/// - opacity: per frame opacity isn't used.
+/// - position: transforms handled in a separate file ? (keyframe animation transfert for camera, pegs, etc...) 
+/// And what about scale/rotation ?
+/// - name: could be used to create empty frames (named "_blank")
+/// </summary>
+/// <param name="jsonLayer"></param>
+
+void OCAIo::OCAInputData::importOcaFrame(const QJsonObject &jsonFrame,
+                                         TXshSimpleLevel *sl) {
+  // doesExists (QFileInfo.exists()) should skip empty frames ! but...
+  TFrameId fid(jsonFrame["frameNumber"].toInt());
+  if (jsonFrame["name"].toString() == "_blank" ||
+      jsonFrame["fileName"].toString() == "") {
+    TImageP img = sl->createEmptyFrame();
+    sl->setFrame(fid, img);
+    return;
+  }
+  auto path       = TFilePath(m_parentDir.getQString() + "/" +
+                        jsonFrame["fileName"].toString());
+  bool fileExists = TFileStatus(path).doesExist();
+  TImageP img;
+  if (fileExists) {
+    auto imageReader = TImageReaderP(path);
+    try {
+      img = imageReader->load();
+      if (img.getPointer() == 0) img = sl->createEmptyFrame();
+    } catch (const std::string &msg)
+    {
+      DVGui::error(QObject::tr("importOcaFrame load failed : ") +
+                   path.getQString());
+      img = sl->createEmptyFrame();
+      throw TException(msg);
+    }
+    if (img->getType() != TImage::RASTER) {
+      DVGui::warning(
+          QObject::tr("importOcaFrame skipping unimplemented image type : ") +
+          jsonFrame["name"].toString());
+    }
+    auto info = imageReader->getImageInfo();
+    DVGui::info(QObject::tr("importOcaFrame : ") + path.getQString() +
+                " width: " + QString::number(info->m_lx) +
+                " height: " + QString::number(info->m_ly) 
+                + " bits: " + QString::number(info->m_bitsPerSample) );
+
+  } else {
+    DVGui::error(QObject::tr("importOcaFrame file not found! : ") + path.getQString());
+    img = sl->createEmptyFrame();
+  }
+  sl->setFrame(fid, img);
+  if (!Preferences::instance()->isImplicitHoldEnabled()) {
+      //do something with the image duration ?
```

Comment:
This doesn't need to be done here.  Image duration needs to be handled only when loading into the xsheet